content/blog/20241214_fluent-package-v5.2.0-has-been-released.md

```diff
@@ -0,0 +1,143 @@
+# fluent-package v5.2.0 has been released
+
+Hi users!
+
+We have released fluent-package [v5.2.0](https://github.com/fluent/fluent-package-builder/releases/tag/v5.2.0).
+fluent-package is a stable distribution package of Fluentd. (successor of td-agent v4)
+
+This release is a new release of v5.2 series.
+
+## Changes from fluent-package v5.1.0
+* Support upgrade fluentd service with zero-downtime
+* Update ruby to 3.2.6
+* Update bundled Fluentd to v1.18.0
+* Update bundled gems overall
+* Solved possibility of DoS through "NoMemoryError"
+* deb rpm: Fixed to not execute v4 restart migration process unexpectedly
+* msi: set `GEM_HOME`/`GEM_PATH` in fluentd.bat
+* Fixed SIGABORT error with fluent-plugin-systemd
+
+## Support upgrade fluentd service with zero-downtime
+Previously, when upgrading the fluent-package, you had to stop the fluentd service,
+install the new version of the fluent-package, then install any necessary plugins,
+and finally restart the fluentd service.
+
+Starting from fluent-package v5.2.0,
+we have introduced the zero-downtime restart feature that automatically installs the required plugins during the upgrade process
+and allows you to upgrade Fluentd without interrupting the logging service.
+
+The zero-downtime restart feature can be configured by `FLUENT_PACKAGE_SERVICE_RESTART` environemnt variable.
+Please refer to following section for more details.
+
+* Add `FLUENT_PACKAGE_SERVICE_RESTART` environmental variable
+  * This section explains `FLUENT_PACKAGE_SERVICE_RESTART` environment variable.
+* Automate Plugin Install for Update on Demand
+  * This section explains the mechanism that install the plug-ins automatically during the upgrade process.
+
+### Add `FLUENT_PACKAGE_SERVICE_RESTART` environmental variable
+We have introduced `FLUENT_PACKAGE_SERVICE_RESTART` environment variable to configure the zero-downtime restart feature.
+The variable exists in the following file:
+
+* RPM: `/etc/sysconfig/fluentd`
+* DEB: `/etc/default/fluentd`
+
+Example:
+
+```
+FLUENT_PACKAGE_OPTIONS=""
+# Control method to upgrade service (auto/manual) restart
+FLUENT_PACKAGE_SERVICE_RESTART=auto
+```
+
+#### **auto (default)**
+The service automatically restarts with zero-downtime restart feature when all of the following conditions are met:
+
+1. The service was active before updating.
+2. The installed and upgrading versions are required 5.2.0 or higher (both sides must support this feature).
+
+The plugins will be automatically reinstalled if needed if your environment has an active online connection.
+For more details, please refer to "Automate Plugin Install for Update on Demand".
+
+**Note:** If the service was active before the update, it will not restart automatically. You need to restart the service manually.
```

Comment:
> **Note:** If the service was active before the update, it will not restart automatically. You need to restart the service manually.

Is not this an explanation for `manual`?

Reply:
indeed